flow/scripts/detail_route.tcl

```diff
@@ -69,4 +69,8 @@ if { [env_var_exists_and_non_empty POST_DETAIL_ROUTE_TCL] } {
 
 check_antennas -report_file $env(REPORTS_DIR)/drt_antennas.log
 
+if {![design_is_routed]} {
+  error "Design has unrouted nets."
```

Comment:
would this signify a bug in OpenROAD or a problem with the design?

If it is the latter, won't it be useful to save & look at DRC errors?

Reply:
We had a flow bug that left some nets unrouted.  It is more of a sanity check.

Reply:
The "error" is the clue that it is more of a problem with OpenROAD than the design?

Reply:
In this the flow but yes.

Reply:
Interesting... I haven't seen "asserts" in the scripts that I can recall. I'm curious why it is exposed in .tcl instead of handled in C++. What is the advantage to scripting it?

Reply:
Again - there is no error in OR.  It is a flow error.

Reply:
Still leaves me scratching my head.

I guess I will wait until I see one; if I never see one, it hardly matters. If I do, I'll learn more about it.

Reading the code I have no idea what to do if I see this error. Do I file a github issue? Do I look at the floorplan/density, etc.?

I don't know what distinguishes, exactly, a "flow error" from a "bug".

If there are DRC errors, then mostly, these days, the flow parameters/design has to be tweaked. DRC errors recently has signified a problem earlier in the flow, like macro placement.